app/processor/src/detection_strategy.py

```
@@ -61,22 +78,48 @@ def is_valid_detection(self, bbox: List[float], conf: float, min_confidence: flo
-             self.logger.info(f'Initializing with regional species filters: {self.regional_species}')
-             self.classes = [id for id, label in self.model.names.items() if any(
-                reg_species in label for reg_species in self.regional_species)]
-             
-             # Log the actual class names that are enabled
-             enabled_classes = [self.model.names[id] for id in self.classes]
-             self.logger.info(f'Regional species filters active: {len(self.classes)} classes enabled.')
-             self.logger.info(f'Enabled classes: {enabled_classes}')
+            self.logger.info(f'Initializing with regional species filters: {self.regional_species}')
+            self.classes = [
+                id
+                for id, label in self.model.names.items()
```

Ruff (0.15.7)

[error] Variable `id` is shadowing a Python builtin

(A001)